corehq/apps/locations/models.py

```diff
@@ -805,6 +813,24 @@ def _unassign_users_from_location(domain, location_id):
             user.unset_location_by_id(location_id, fall_back_to_next=True)
 
 
+def get_case_sharing_groups_for_locations(locations, for_user_id=None):
+    # safety check to make sure all locations belong to same domain
+    assert len(set([l.domain for l in locations])) < 2
+
+    for location in locations:
+        if location.location_type.shares_cases:
+            yield location.case_sharing_group_object(for_user_id)
+
+    location_ids = [l.pk for l in locations if l.location_type.view_descendants]
+    descendants = []
+    if location_ids:
+        where = Q(domain=locations[0].domain, parent_id__in=location_ids)
+        descendants = SQLLocation.objects.get_queryset_descendants(where).filter(
```

Comment:
It doesn't really matter since they're essentially the same, but technically this should call `SQLLocation.objects.get_descendants()` (you're not passing in a `QuerySet`).